docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,297 @@
 
-When multiple services need to be started together, they're started in order according to the `before` and `after` configuration, waiting 1 second for each to ensure the command doesn't exit too quickly. The `before` option is a list of services that this service must start before (it may or may not `require` them). Or if it's easier to specify this ordering the other way around, `after` is a list of services that this service must start after.
+## Start order
+
+When multiple services need to be started together, they're started in order according to the `before` and `after` configuration (see more detail in [Layer specification](../reference/layer-specification.md)), waiting for 1 second for each to ensure the command doesn't exit too quickly.
+
+The `before` option is a list of services that this service must start before (it may or may not `require` them). Or if it's easier to specify this ordering the other way around, `after` is a list of services that this service must start after.
 
-Note that currently, `before` and `after` are of limited usefulness, because Pebble only waits 1 second before moving on to start the next service, with no additional checks that the previous service is operating correctly.
+```{note}
+Currently, `before` and `after` are of limited usefulness, because Pebble only waits 1 second before moving on to start the next service, with no additional checks that the previous service is operating correctly.
 
 If the configuration of `requires`, `before`, and `after` for a service results in a cycle or "loop", an error will be returned when attempting to start or stop the service.
+```
+
+## Putting it together
+
+With the service dependencies and start order features, let's reconfigure our web app layer to:
+
+`001-my-web-app-layer.yaml`:
+
+```yaml
+services:
+  database:
+    override: replace
+    summary: database
+    command: python3 -m http.server 3306
+    startup: enabled
+  backend-server:
+    override: replace
+    summary: backend
+    command: python3 -m http.server 8081
+    startup: enabled
+    requires:
+      - database
+    after:
+      - database
+  frontend-server:
+    override: replace
+    summary: frontend
+    command: python3 -m http.server 8080
+    startup: enabled
+    requires:
+      - backend-server
+    after:
+      - backend-server
+```
+
+If we try to start it now:
+
+```{terminal}
+   :input: pebble run
+2024-06-28T02:17:23.347Z [pebble] Started daemon.
+[backend-server database frontend-server]
+2024-06-28T02:17:23.353Z [pebble] POST /v1/services 2.613042ms 202
+2024-06-28T02:17:23.356Z [pebble] Service "backend-server" starting: python3 -m http.server 8081
+2024-06-28T02:17:24.363Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:17:25.370Z [pebble] Service "frontend-server" starting: python3 -m http.server 8080
```

Comment:
With the configuration above, I would expect to see database, backend-server, then frontend-server -- in that order due to the `after` configuration. That's what I get locally -- was this a copy 'n' paste error?